index.d.ts

```diff
@@ -121,7 +121,11 @@ declare namespace fastifyGQL {
         info: object,
         next: (result: boolean) => void
       ) => void
-    }
+    },
+    /**
+     * Enable federation support so the service can be deployed behind an Apollo Gateway
+     */
+    enableFederation?: boolean
```

Comment:
How about we call this `federationMetadata`? I think it reflects more the fact that it will serve the federation metadata itself.